pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -39,6 +39,42 @@ const (
 	VirtualInboundListenerName = "virtualInbound"
 )
 
+var (
+	// DeprecatedFilterNames is to support both canonical filter names
+	// and deprecated filter names for backward compatibility. Istiod
+	// generates canonical filter names.
+	DeprecatedFilterNames = map[string]string{
```

Comment:
@ramaraochavali this is great.
1. We need to add release notes for this pr.

2. 

I am wondering if we should warn when we see the use of older names ? The problem is no one looks at istiod logs unless something goes wrong. We can have an env var for validation that will reject all Deprecated names.
I don’t think we want to support old names for ever.

@kyessenov now that wellknown names are in the api, can they be treated as such or were they put there as coding convenience?

Reply:
Well known names are not part of API. I've been telling people to stop using them.

Reply:
This definitely needs documentation in EnvoyFilter since it changes its behavior.

Reply:
I am planning to add follow-up with doc updates to EnvoyFilter after this is merged.